src/app/core/common-components/entity-field-edit/entity-field-edit.component.ts

```diff
@@ -67,14 +76,23 @@ export class EntityFieldEditComponent<T extends Entity = Entity>
   }
 
   private updateField() {
-    if (!this.entity?.getConstructor()) {
-      this._field = !!this.field ? toFormFieldConfig(this.field) : undefined;
+    if (!this.field) {
+      this._field = undefined;
       return;
     }
 
-    this._field = this.entityFormService.extendFormFieldConfig(
-      this.field,
-      this.entity.getConstructor(),
-    );
+    this._field = toFormFieldConfig(this.field);
+
+    if (this.entity?.getConstructor()) {
+      this._field = this.entityFormService.extendFormFieldConfig(
+        this.field,
+        this.entity.getConstructor(),
+      );
+    } else {
+      // add editComponent (because we cannot rely on the entity's schema yet for a new field)
+      this._field.editComponent =
+        this._field.editComponent ??
+        this.entitySchemaService.getComponent(this._field, "edit");
+    }
```

Comment:
```suggestion
    if (this.entity?.getConstructor()) {
      this._field = this.entityFormService.extendFormFieldConfig(
        this.field,
        this.entity.getConstructor(),
      );
    } else {
      this._field = toFormFieldConfig(this.field);
      // add editComponent (because we cannot rely on the entity's schema yet for a new field)
      this._field.editComponent =
        this._field.editComponent ??
        this.entitySchemaService.getComponent(this._field, "edit");
    }
```